rosbag2/src/rosbag2/writer.cpp

```diff
@@ -114,6 +114,8 @@ void Writer::remove_topic(const TopicMetadata & topic_with_type)
 
   if (topics_names_to_info_.erase(topic_with_type.name) > 0) {
     storage_->remove_topic(topic_with_type);
+  } else {
+    throw std::runtime_error("Tried to remove a non-existing topic");
```

Comment:
it'd be preferable to have this contain the topic name.

```c++
std::stringstream ss;
ss << "failed to remove non-existing topic \"" << topic_with_type.name.c_str() << "\"";
throw std::runtime_error(ss.str());
```

Note: unless I directly create code suggestions you can merge, the code in my comments is usually not production-ready, it's usually just enough so the reader can understand the intent, hence this additional comment :)